Library/OcDeviceMiscLib/SetResizableBar.c

```diff
@@ -309,8 +377,241 @@ SetResizableBarOnDevice (
+                                  );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "OCDM: RBAR caps cannot be read - %r\n", Status));
+    return EFI_UNSUPPORTED;
+  }
+
+  Status = PciRootBridgeIo->Pci.Read (
+                                  PciRootBridgeIo,
+                                  EfiPciWidthUint32,
+                                  PciAddrOffset (PciAddress, OFFSET_OF (PCI_TYPE00, Device.Bar)),
+                                  PCI_MAX_BAR,
+                                  (VOID *)OldBar
+                                  );
+  if (EFI_ERROR (Status)) {
+    ZeroMem (OldBar, sizeof (OldBar));
+  }
+
+  DEBUG ((
+    DEBUG_INFO,
+    "OCDM: Old BAR %08X %08X %08X %08X %08X %08X - %r\n",
+    OldBar[0],
+    OldBar[1],
+    OldBar[2],
+    OldBar[3],
+    OldBar[4],
+    OldBar[5],
+    Status
+    ));
+
+  for (Index = 0; Index < ResizableBarNumber; Index++) {
+    //
+    // When the bit of Capabilities Set, indicates that the Function supports
+    // operating with the BAR sized to (2^Bit) MB.
+    // Example:
+    // Bit 0 is set: supports operating with the BAR sized to 1 MB
+    // Bit 1 is set: supports operating with the BAR sized to 2 MB
+    // Bit n is set: supports operating with the BAR sized to (2^n) MB
+    //
+    // Reference values for RX 6900 with two resizable BARs.
+    // Disabled values:
+    //   Resizeable Bar Capability [1]
+    //     ResizableBarCapability         0007F000
+    //     ResizableBarControl            0840
+    //   Resizeable Bar Capability [2]
+    //     ResizableBarCapability         00001FE0
+    //     ResizableBarControl            0102
+    // Enabled values:
+    //   Resizeable Bar Capability [1]
+    //     ResizableBarCapability         0007F000
+    //     ResizableBarControl            0E40
+    //   Resizeable Bar Capability [2]
+    //     ResizableBarCapability         00001FE0
+    //     ResizableBarControl            0802
+    //
+    NewCapabilities = Capabilities = LShiftU64 (Entries[Index].ResizableBarControl.Bits.BarSizeCapability, 28)
+                                     | Entries[Index].ResizableBarCapability.Bits.BarSizeCapability;
+
+    //
+    // Restrict supported BARs to specified value.
+    //
+    NewCapabilities &= PCI_BAR_CAP_LIMIT (Size);
+
+    //
+    // Disable bits higher than current as we are not allowed to increase bar size
+    // more than we already have.
+    //
+    if (!Increase) {
+      NewCapabilities &= PCI_BAR_CAP_LIMIT (Entries[Index].ResizableBarControl.Bits.BarSize);
+    }
+
+    //
+    // If requested BAR size is too low, choose the lowest available BAR size.
+    //
+    if (  (NewCapabilities == 0)
+       && (Entries[Index].ResizableBarControl.Bits.BarSize > (UINT32)Size))
+    {
+      Bit = LowBitSet64 (Capabilities);
+    } else {
+      Bit = HighBitSet64 (NewCapabilities);
+    }
+
+    DEBUG ((
+      DEBUG_INFO,
+      "OCDM: RBAR %u/%u supports 0x%Lx, sizing %u inc %d results setting from %u to %d\n",
```

Comment:
Maybe change the comment to be able to distinguish PciIo version from RbIo version from the log?

Reply:
Already it gets printed that PciRootBridgeIo or PciIo is being used so I thought it wouldn't needed to add it here

Reply:
Right, never mind.